i6_models/parts/frontend/vgg_act.py

```diff
@@ -0,0 +1,200 @@
+        """
+        :param model_cfg: model configuration for this module
+        """
+        super().__init__()
+
+        model_cfg.check_valid()
+
+        conv_padding = (
+            model_cfg.conv_padding if model_cfg.conv_padding is not None else _get_padding(model_cfg.conv_kernel_size)
+        )
+        pool1_padding = model_cfg.pool1_padding if model_cfg.pool1_padding is not None else 0
+        pool2_padding = model_cfg.pool2_padding if model_cfg.pool2_padding is not None else 0
+
+        self.include_linear_layer = True if model_cfg.out_features is not None else False
+        assert not (
+            (model_cfg.linear_input_dim is not None) != (model_cfg.out_features is not None)
+        ), "please set input and output dim of the linear layer"
+
+        self.conv1 = nn.Conv2d(
+            in_channels=model_cfg.in_features,
+            out_channels=model_cfg.conv1_channels,
+            kernel_size=model_cfg.conv_kernel_size,
+            padding=conv_padding,
+        )
+        self.conv2 = nn.Conv2d(
+            in_channels=model_cfg.conv1_channels,
+            out_channels=model_cfg.conv2_channels,
+            kernel_size=model_cfg.conv_kernel_size,
+            padding=conv_padding,
+        )
+        self.pool1 = nn.MaxPool2d(
+            kernel_size=model_cfg.pool1_kernel_size,
+            stride=model_cfg.pool1_stride,
+            padding=pool1_padding,
+        )
+        self.conv3 = nn.Conv2d(
+            in_channels=model_cfg.conv2_channels,
+            out_channels=model_cfg.conv3_channels,
+            kernel_size=model_cfg.conv_kernel_size,
+            padding=conv_padding,
+        )
+        self.conv4 = nn.Conv2d(
+            in_channels=model_cfg.conv3_channels,
+            out_channels=model_cfg.conv4_channels,
+            kernel_size=model_cfg.conv_kernel_size,
+            padding=conv_padding,
+        )
+        self.pool2 = nn.MaxPool2d(
+            kernel_size=model_cfg.pool2_kernel_size,
+            stride=model_cfg.pool2_stride,
+            padding=pool2_padding,
+        )
+        self.activation = model_cfg.activation
+        if self.include_linear_layer:
+            self.linear = nn.Linear(
+                in_features=model_cfg.linear_input_dim,
+                out_features=model_cfg.out_features,
+                bias=True,
+            )
+
+    def forward(
+        self, tensor: torch.Tensor, sequence_mask: Optional[torch.Tensor]
+    ) -> Tuple[torch.Tensor, Optional[torch.Tensor]]:
+        """
+        T might be reduced to T' or T'' depending on stride of the layers
+
+        stride is only allowed for the pool1 and pool2 operation.
+        other ops do not have stride configurable -> no update of mask sequence required
+
+        :param tensor: input tensor of shape [B,T,F]
+        :param sequence_mask: the sequence mask for the tensor
+        :return: torch.Tensor of shape [B,T",F'] and the shape of the sequence mask
+        """
+        # and add a dim
```

Comment:
I think we should add a check here that the input feature dim is as configured, sth like:
```suggestion
        assert tensor.shape[-1] == self.cfg.in_features
        # and add a dim
```